strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/io/MavenMetadataExpiredRepositoryPathHandler.java

```diff
@@ -50,74 +60,31 @@ public boolean supports(final RepositoryPath repositoryPath)
     public void handleExpiration(final RepositoryPath repositoryPath)
             throws IOException
     {
-        Decision refetchMetadata = determineMetadataRefetch(repositoryPath,
-                                                            EncryptionAlgorithmsEnum.SHA1);
-        if (refetchMetadata == I_DONT_KNOW)
-        {
-            refetchMetadata = determineMetadataRefetch(repositoryPath,
-                                                       EncryptionAlgorithmsEnum.MD5);
-        }
-        if (refetchMetadata == NO_LEAVE_IT)
+        MetadataExpirationStrategy metadataExpirationStrategy = getMetadataStrategy(repositoryPath);
+        MetadataExpirationStrategy.Decision refetchMetadata = metadataExpirationStrategy.decide(repositoryPath);
+
+        if (refetchMetadata == USABLE)
         {
-            // checksums match - do nothing
-            logger.debug("Local and remote checksums match - no need to re-fetch maven-metadata.xml.");
             return;
         }
-        if (refetchMetadata == I_DONT_KNOW)
-        {
-            logger.debug("maven-metadata.xml will be re-fetched. Checksum comparison process was not helpful.");
-        }
-        if (refetchMetadata == YES_FETCH)
-        {
-            logger.debug("maven-metadata.xml will be re-fetched. Checksums differ.");
-        }
         proxyRepositoryArtifactResolver.fetchRemoteResource(repositoryPath);
     }
 
-    private Decision determineMetadataRefetch(final RepositoryPath repositoryPath,
-                                              final EncryptionAlgorithmsEnum checksumAlgorithm)
-            throws IOException
+    private MetadataExpirationStrategy getMetadataStrategy(final RepositoryPath repositoryPath)
```

Comment:
we should create some common utility for this method, but we can do it in a follow-up task

Reply:
Reminder that it is still a valid comment if we merge it

Reply:
@cbrianhill : 

Would you mind adding a follow-up task for this and backlinking it to this one? Thanks! :)